File: code/Cracking/HardProblem/l3/MuddyRoad2--Can't understand/solve/cpp/MuddyRoad2Runner.cpp
#include <topcoder>
using namespace TopCoder;
#include "MuddyRoad2.cpp"
#include <fstream>
using namespace std;

int main(int argc, char const *argv[]) {
    try {
        ifstream ifs(argv[1]);
        int N; read(ifs, N); next(ifs);
        int muddyCount; read(ifs, muddyCount);
        ifs.close();

        ofstream ofs(argv[2]);
        show(ofs, theCount(N, muddyCount));
        ofs.close();
    } catch (exception &e) {
        cerr << e.what() << endl;
    }
    return 0;
}

File: code/Cracking/HardProblem/l3/MuddyRoad2--Can't understand/solve/java/MuddyRoad2Runner.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class MuddyRoad2Runner {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int N = (Integer) reader.next(Integer.class);
        reader.next();
        
        int muddyCount = (Integer) reader.next(Integer.class);
        reader.close();

        MuddyRoad2 solver = new MuddyRoad2();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.theCount(N, muddyCount));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
